### ISSUE TITLE
1.0+ way to serve files from /node_modules/ ?

### ISSUE THREAD

**Issue by markerikson** [davezuko/react-redux-starter-kit#477]
I'm planning to use Webpack along with the Cesium.js 3D globe library.  Per https://github.com/mmacaula/cesium-webpack , this is a somewhat non-trivial exercise.  The recommendations on that page are to explicitly copy the pre-built version of Cesium and its various assets into some sort of public folder.  However, since both the minified and unminified versions of Cesium come included in the Cesium NPM package, I would prefer to just serve those directly out of their location under `/node_modules/cesium/Build/` instead, with some logic for selecting the right subfolder (`/Cesium/Cesium.js`, or `/CesiumUnminified/Cesium.js`).  

My original test project was RRSK v0.17+, which was still using Babel 5 and Webpack-Dev-Server.  I was able to convince it to let me serve files out of `/node_modules/cesium/` with the following edits:

* `build/webpack-dev-server.js`: changed `contentBase` from `paths.project(config.get('dir_src'))` to `"."`

* `build/webpack/development_hot.js`: adding
```
var CESIUM_BASE_URL = "./node_modules/cesium/Build/CesiumUnminified/";

webpackConfig.plugins.push(new webpack.DefinePlugin({
  $CESIUM_BASE_URL : JSON.stringify("/node_modules/cesium/Build/CesiumUnminified/")
}));

var cesiumPath = config.get("utils_paths").project("node_modules/cesium/Build/CesiumUnminified/");

webpackConfig.resolve.alias.cesium = cesiumPath;
webpackConfig.module.noParse.push(cesiumPath + "/Cesium.js");
```

Also made some edits to `build/_base.js` to allow Cesium to load correctly, etc.

However, in RRSK 1.0, I can't seem to convince the server to let me request static files under `/node_modules/`.  I'm able to request Cesium.js itself by using the same alias and such, but I can't figure out how to allow any other requests to work.  I've tried editing `/server/middleware/webpack-dev.js` and modifying `contentBase`, as well as the `app.use(express.static())` expression in `/server/app.js`.  No luck so far.

I realize that what I'm trying to do is a bit esoteric, but any suggestions on a way to expose static files that are located under `/node_modules/cesium/` with RRSK 1.0+?

**Comment by markerikson** [comment]
Follow-up on that stuff for `development_hot.js`: looks like I was a bit redundant, but the intent there was to add a define value that can use on the client-side to help configure Cesium correctly, letting it how to build URLs for its assets.  The alias was needed to let me `import Cesium from "cesium"`, and the noParse was a suggestion for getting a big library pulled in without having Webpack try to parse it fully.

**Comment by markerikson** [comment]
Turns out I was doing a couple things wrong.  First, I was editing the Express `app.use(express.static())` reference inside the non-HMR clause of `/server/app.js`.  Since I was running in development mode, those edits were effectively useless.

Once I figured that out, I was able to successfully serve up assets from `node_modules` by adding this as unconditional logic in app.js:

```js
let isDev = config.globals["__DEV__"];

let prebuiltCesiumFolder = isDev ? "CesiumUnminified" : "Cesium";

let cesiumPackagePath = "node_modules/cesium/Build/" + prebuiltCesiumFolder;
let absoluteCesiumPath = paths.base(cesiumPackagePath);

app.use('/cesium', express.static(absoluteCesiumPath));
```


So, the ultimate answer was "do `app.use(express.static())`, but actually make sure the code runs".
